### DISCUSSION TITLE
How to set priority for devices?

### DISCUSSION THREAD

**Discussion by arnkit** [jmcollin78/solar_optimizer#69] [Q&A]
Hi,

I'm a happy user of solar optimizer - works great!

I was wondering if there is any way of setting a priority to devices. The main idea is, that devices are switched of according to their priority. For example
I have have a dehumidifier, and a electric floor heating - and I always want the dehumidifier to run first (priority 1), while other devices may be switched on if afterwards enough solar power is available. Same situation would apply to devices with the same amount of watts consumed.

Thanks

**Comment by jmcollin78** [comment]
Hello,

This not possible out of the box, because the optimisation algo is searching the best combination of 'on'/'off' to minimise the "cost" function.

BUt, you could certainly do something with check_enable template, to disable the lower priority device if a higher priority is enabled and not active. This should work.

**Reply by wimpie007** [reply]
mmm, let say i have 3 on/off devices, 1 with 500W (lowest prio), second one with 1000W (middle prio), and third one with 2000W load. (max prio)
I want overal priority for the 2000W load IF there is solar power enough.  So if solar power>2000 for x time, i want it to switch on, and i want the 1. and 2. devices to be switched off.

if solar power is for example 500 watt: turn 1. on.
1200W, i want 2. to be turned on.  
1500 watt-> 1. and 2. 
2000 -> 3.
2500 -> 1. and 3.
and so on...
Without 'priority' mode, thats very hard to accomplish... (i've tried ;) )

Costs: i just want to use my 'free' solar power to the max...  so it is not needed for any device to run 'at least x hours a day'
I think many people have this problem...

**Reply by jmcollin78** [reply]
Hello you don't need SO to do that but a simple automation. Priority is totally in contradiction with the principles of SO.
Why just not let the algorithm find the best solution ? It will ensure to use the solar power to the max. That is exactly the purpose

**Reply by wimpie007** [reply]
Hi Jean-Marc!
but what if you have equal 'cost'?
2 devices that use same amount of watt?
which one will be switched on?

or even a scenario with
A 700W
B 1500W
C 2200W ( highest priority)
none if the loads must be on for a minimum of time.

if solar power= 2200 then the optimal 'cost' solution is running A+B  instead of running C.
Thats not what i want that moment ...

PS dont get me wrong, you're doing a great job! just start to think SO is not a solution for these kind of problems.(there seems to be no real solution out there also).

**Reply by jmcollin78** [reply]
There is no simple way with this algorithm (https://en.wikipedia.org/wiki/Simulated_annealing) to integrate priority.

I will have to integrate the priority into the evaluation function. What is more important the priority or the cost function ?
Suppose for one point more you respect the priority, should we take this result ?

Sp this is too much complexity and the result (the cost function will be the same at the end). All I will have is to potentially degrade the objective (optimize the produced power consumption).

So sorry, I will not integrate this.

**Reply by wimpie007** [reply]
no problem, thanks for replying.
I will look into another solution.
Thanks anyway!
regards,
